http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/monitor/LeoPubsubMessageSubscriber.scala

```diff
@@ -528,9 +532,105 @@ class LeoPubsubMessageSubscriber[F[_]: Timer: ContextShift](
       )
     } yield ()
 
-  def handleCreateAppMessage(msg: CreateAppMessage): F[Unit] = F.unit
+  def handleCreateAppMessage(msg: CreateAppMessage)(
+    implicit ev: ApplicativeAsk[F, AppContext]
+  ): F[Unit] = {
+    val createClusterAndNodepool =
+      for {
+        _ <- msg.cluster.fold(F.unit)(createClusterMessage => gkeInterp.createCluster(createClusterMessage, msg.appId))
+        _ <- gkeInterp.createNodepool(msg.nodepoolId, msg.appId)
```

Comment:
We won't _always_ need to create a nodepool during a `createApp` message -- wondering if we could represent that in a similar way as creating a cluster.

Reply:
See above comment in terms of the new nodepool concerns. I'd prefer to do all the refactoring at once since we'll need a moderate amount of new logic